npm/debugTools/dataplane/parser_test.go

```diff
@@ -0,0 +1,208 @@
+package dataplane
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/npm/util"
+)
+
+func TestParseIptablesObjectFile(t *testing.T) {
+	p := &Parser{}
+	p.ParseIptablesObjectFile(util.IptablesFilterTable, "../testFiles/iptableSave")
+}
+
+func TestParseIptablesObject(t *testing.T) {
+	p := &Parser{}
+	p.ParseIptablesObject(util.IptablesFilterTable)
+}
+
+func TestParseLine(t *testing.T) {
+	type test struct {
+		input    string
+		expected []byte
+	}
+
+	p := &Parser{}
+
+	// line with no left or right space
+	testL1 := "-A AZURE-NPM -m mark --mark 0x3000 -m comment --comment TEST -j AZURE-NPM-ACCEPT"
+	// line with left space
+	testL2 := "      -A AZURE-NPM -m mark --mark 0x3000 -m comment --comment TEST -j AZURE-NPM-ACCEPT"
+	// line with right space
+	testL3 := "-A AZURE-NPM -m mark --mark 0x3000 -m comment --comment TEST -j AZURE-NPM-ACCEPT       "
+	// line with left and right space
+	testL4 := "        -A AZURE-NPM -m mark --mark 0x3000 -m comment --comment TEST -j AZURE-NPM-ACCEPT       "
+
+	expectByteArray := []byte("-A AZURE-NPM -m mark --mark 0x3000 -m comment --comment TEST -j AZURE-NPM-ACCEPT")
+
+	tests := []test{
+		{
+			input:    testL1,
+			expected: expectByteArray,
+		},
+		{
+			input:    testL2,
+			expected: expectByteArray,
+		},
+		{
+			input:    testL3,
+			expected: expectByteArray,
+		},
+		{
+			input:    testL4,
+			expected: expectByteArray,
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.input, func(t *testing.T) {
+			actualLine, _ := p.parseLine(0, []byte(tc.input))
+			if equal := bytes.Compare(expectByteArray, actualLine); equal != 0 {
+				t.Errorf("expected '%+v', got '%+v'", tc.expected, actualLine)
```

Comment:
nitpick: it seems idiomatic of go is to print failed message with the order of "got" and "expected".
Refer to https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures